python/doc/source/memory.rst

```diff
@@ -70,11 +74,42 @@ required, and such conversions are also zero-copy:
 
    memoryview(buf)
 
-.. _io.native_file:
-
-Native Files
+Memory Pools
 ------------
 
+All memory allocations and deallocations (like ``malloc`` and ``free`` in C)
+are tracked in an instance of ``arrow::MemoryPool``. This means that we can
+then precisely track amount of memory that has been allocated:
+
+.. ipython:: python
+
+   pa.total_allocated_bytes()
+
+PyArrow uses a default built-in memory pool, but in the future there may be
+additional memory pools (and subpools) to choose from.  Let's allocate
```

Comment:
```suggestion
additional memory pools (and subpools) to choose from. Let's allocate
```

Reply:
Putting two spaces after a period is a [common convention](https://devguide.python.org/documenting/#use-of-whitespace). I'm not sure whether we have any guidelines on that. @wesm @xhochy

Reply:
It seems like both conventions are used in the rst docs.

Reply:
AP Stylebook for journalism today is 1 space. 2 spaces seems to be a remnant of the typewriter era